java/fury-core/src/main/java/org/apache/fury/serializer/collection/ChildContainerSerializers.java

```diff
@@ -210,11 +216,14 @@ public Map newMap(MemoryBuffer buffer) {
     }
 
     @Override
-    public T copy(T originMap) {
-      if (objectSerializer == null) {
-        objectSerializer = new ObjectSerializer<>(fury, type, false);
+    public Map newMap(Map originMap) {
```

Comment:
Why can't we just delegate copy to objectSerializer? I think we should follow same pattern as serialization did

Reply:
I see, objectSerializer can set fields for current type